ci/requirements/environment.yml

```diff
@@ -46,3 +46,5 @@ dependencies:
   - toolz
   - typing_extensions
   - zarr
+  - pip:
+    - git+https://github.com/xarray-contrib/datatree
```

Comment:
is there a reason why we're installing from github here?

Reply:
Because I want to see if [this commit to datatree](https://github.com/xarray-contrib/datatree/commit/927749a7702761491461f9bdfa8a0c1fbc244d85) fixes the mypy issue